internal/querycoordv2/checkers/balance_checker.go

```diff
@@ -141,6 +116,11 @@ func (b *BalanceChecker) balanceReplicas(replicaIDs []int64) ([]balance.SegmentA
 }
 
 func (b *BalanceChecker) Check(ctx context.Context) []task.Task {
+	// scheduler is handling segment task, skip
+	if b.scheduler.GetSegmentTaskNum() != 0 {
```

Comment:
this may be not a good strategy, the scheduler may be processing tasks at most of time, it's OK to add a new task to it